### PR TITLE
Radio station broadcast audio archives - lore

### PR DESCRIPTION
**Pull request by nexusmrsep**
```SUMMARY: Features "Radio station broadcast audio archives"```

#### Purpose of change
We have radio towers and radio stations. Radio stations record their (and other) broadcast for archival purposes, and in the days of Cataclysm many of them have been re-purposed for military/emergency broadcast.  This must mean that many interesting recordings had been archived in that short period, including:
- military broadcast,
- police broadcast,
- civilian self-defense forces broadcast, including firefighters, medics and other emergency services,
- local/central government broadcast,
- normal broadcast (going haywire as time goes by, news getting more dark and unrelenting),
- etc.

So the main aim here is to add another channel to introduce lore to the game, this time focused on what was happening "in the air" before and during Cataclysm.

I plan this as a community participation effort, so I deliver the infrastructure, but the ideas for the content itself (radio messages, broadcast snippets) will be gathered from the community, mainly Discourse and Reddit.

#### Describe the solution
Added `radio_archive` computer action to computer consoles, that plays random snippets with "radio_archive" tag. Enabled this action from the radio station computer consoles. Since some of the broadcast has restricted access, possibly enforced by military, there is a limit of what you can squeeze out of such console.

#### Describe alternatives you've considered
Broadcasting it directly to air (received by a radio), but it would seem odd to listen to some old archived  broadcast this way.

#### Additional context
Many thanks for the respective contributors of snippets who supplied their ideas here:
- [Reddit](https://www.reddit.com/r/cataclysmdda/comments/bo6cjd/radio_archives_community_project/)
- [Discourse](https://discourse.cataclysmdda.org/t/audio-archives-community-project/20055)
- this thread

### CODE REVIEW

**Comment by Ilysen** [comment]
Some cool low-hanging fruit would be a reference to The Phenomenon, which was a reddit writing prompt that developed into a bigger short story.

```
Please remain in your homes, if you are not at home, find shelter immediately. Close all blinds and shades, block out all windows.
Do not look outside.
Do not look at the sky.
Do not make noise.
Your cooperation is vital to your survival. Appointed government personnel will update you shortly.
```

**Comment by kevingranade** [comment]
This pull request has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/audio-archives-community-project/20055/1

**Comment by ZhilkinSerg** [comment]
> Some cool low-hanging fruit would be a reference to The Phenomenon, which was a reddit writing prompt that developed into a bigger short story.

How is it related to Cataclysm? Something from the portals came to Earth?

**Comment by kevingranade** [comment]
I do think some radio stations would be left on broadcast for a while, but I agree it wouldn't be the norm, and a lot of this content would not be accessible that way.

**Comment by I-am-Erk** [comment]
So, there's a good supply of inspiration in the timeline probably: https://cataclysmdda.org/lore-background.html

In there, there's what I'd like to canonize as the last use of the emergency broadcast system:
`The government has fallen. No help is coming, you’re on your own.`

Other ideas:
`You're listening to KDDA from Boston, I'm Jenny Sanders with a special emergency news update.

Road closures by military and police forces have been reported on interstates 90, 91, 93, and 95.  Alternative routes have not been suggested.  Due to dangers on non-interstate roads, it is strongly recommended that you avoid major thoroughfares and population centers when travelling by car.

Those evacuating cities are advised to head to the nearest designated evacuation shelter and await pickup by a FEMA transport.  Do not engage with rioters.  Due to the danger of being mistaken for a rioter, do not approach military or police blockades, even for assistance.  I repeat, do not approach military or police blockades, even for assistance.  Police robots cannot always distinguish between a panicking civilian and a rioter, so you can set yourself apart by avoiding blockades altogether.

If you are not in an evacuation zone, officials highly recommend you pack gear for evacuation.  Make sure to include clean clothes, a blanket, and enough food and water to last a few days.

In other news, the federal government has confirmed that Rhode Island's declaration of secession is official, although the US government does not acknowledge its validity. Our sources haven't been able to  determine the extent of hostilities in the area, but there are rumors of border violence initiated by the Rhode Island militia.`

I've got a friend in radio, I'll ask him for ideas for what might be looping on repeat after an apocalypse.

**Comment by Ilysen** [comment]
@ZhilkinSerg The actual phenomenon itself might not be a thing, but I thought it would be an interesting reference. During the end, conspiracy theorists were probably having a field day about, who, and where everything was about (and my headcanon is that the clouds are a constant, sickly green), so somebody trying to sound official and saying the sky is going to kill everyone doesn't seem too outlandish.

(It might just be too jarring, though.)

**Comment by Mecares** [comment]
Cant think of anything more awesome than taking some passages from this:
https://casn.berkeley.edu/wp-content/uploads/resource_files/War_of_Worlds_script.pdf

**Comment by I-am-Erk** [comment]
We're our own world, I don't know why we'd be pulling passages from classic sci fi or reddit.

**Comment by nexusmrsep** [comment]
> We're our own world, I don't know why we'd be pulling passages from classic sci fi or reddit.

I agree on not citing other people's works, but those my be a good reference for how emergency radio broadcast is presented.

**Comment by Ilysen** [comment]
We have a ton of references in-game. Quotes for being high on weed draw directly from pop culture, a lot of professions have song lyrics in their descriptions, the Shotgun Arm CBM is a reference to Ghost in the Shell, the Green Fever fliers are a nod to L4D's Green Flu... the list kinda goes on. I don't see any harm in nods to other things!

**Comment by tinukedaya** [comment]
I agree that broadcasting the archives would be weird on it's own, but maybe add a function to the console to enable the broadcast. Then you can listen to it via radio when you feel like.

**Comment by nexusmrsep** [comment]
> I don't see any harm in nods to other things!

Nods and references are ok, if they are part of the culture and are know to the general public, but parsing fragments of books/novels/etc. directly is a no-go. Copyrights, non-original work, inability to refer to the source etc. And what's the problem with inventing some original ideas on that field? There is no need to base on other works, even if they would be great.

**Comment by I-am-Erk** [comment]
OK. The average radio station would be gone after 6 hours to a day, so we don't need too much. It is not too difficult to take a transmitter and run it from a longer term backup, so it might be fun to have KDDA still running for a while, possibly even putting out a few new updates for a little while before cutting out entirely after 2 weeks or so.

After that, we'll probably need an equivalent to Radio Free Wasteland.

**Comment by Fris0uman** [comment]
Wouldn't the different factions set up their own broadcast after a while ?

**Comment by I-am-Erk** [comment]
They *might*. I'm not sure how many of them would want a radio broadcast.

When faction stuff is getting rolling on a larger scale, the Merchants might have a broadcast advertising their trade routes, but their fear of banditry would probably keep them from being too specific. 

Most other factions don't really want attention, I don't know how many would use radio.

I have some pretty good ideas sketched out for Dark Days Radio as a station that sends out gameplay tips and tricks, lore hints and misconceptions, and DJs music though.

**Comment by paulenka-aleh** [comment]
Sounds crazy but I'd love to catch something like Galaxy Radio Station from Fallout 3. Details here: https://fallout.fandom.com/wiki/Fallout_3_radio_stations

Imagine a guy, possibly a prepper, being caught in a cataclysm while at work. He reinforces his radio tower and continues to operate it just 'cause he likes his job most of the time broadcasting his favorite tracklist and from time to time post-cataclysm news :)

**Comment by I-am-Erk** [comment]
That's more or less what I'm thinking with Dark Days Radio. DJ Dustbowl and Sam, the faithful dog, sitting in a bunker with a radio transmitter, communicating with other survivors by ham radio and sending out an FM signal because they can.

`Hey ladies and gentlemen of the apocalypse, you're listening to Dark Days Radio, the last radio station on the Eastern Seaboard, I'm DJ Dustbowl here with our mascot Sam.`

`DJ Dustbowl back, you're listening to Dark Days Radio.  Quick survivor tip: remember, killing a zombie isn't enough.  You've gotta smash that thing to a pulp or chop it into pieces if you don't want it getting back up.  Good news though, you don't need to worry about headshots!  Any major damage will lay a shambler out.`

`You're listening to Dark Days Radio, I'm DJ Dustbowl! In with a quick tip for any survivors listening: one zombie is bad news, two are worse.  Don't go Rambo on me.  Tackle threats in small numbers or through bottlenecks.  You aren't tougher than a zombie but you are smarter, use that to your advantage... or just run away.`

I've been writing up a lot, probably I'll have enough to make a full fleshed out radio station pretty soon.

**Comment by eilaattwood** [comment]
> When faction stuff is getting rolling on a larger scale, the Merchants might have a broadcast advertising their trade routes, but their fear of banditry would probably keep them from being too specific.
> 
> Most other factions don't really want attention, I don't know how many would use radio.

What about some ciphered messages that can, if decrypted, lead to some POIs?

**Comment by Rail-Runner** [comment]
Hope it's useful. This video below shows plenty of military broadcast that was cut from CoD: MW2 game; evac site is being under attack and situation soon gets real bad. Sounds very fitting for CDDA, but someone would have to rewrite the phrases a bit so that it actually fits and wouldn't trigger copyright issues.
https://www.youtube.com/watch?v=-RplHnI7iqI

**Comment by DracoGriffin** [comment]
[Pages 15-17 of this link](https://www.dhs.wisconsin.gov/publications/p0/p00652.pdf) give some good guidelines on how emergency broadcasts are likely to be written, and could be a useful guideline somewhere for future contributions for radio-related things.

**Comment by nexusmrsep** [comment]
This is ready at this point. Thank you for your ideas.

File: data/json/snippets.json [hunk 1]
```diff
@@ -2292,6 +2292,34 @@
       "XE037 revivification has reached critical levels, which are rapidly overwhelming the available military and regional police assets' ability to stop.  Every team in the field has encountered hostile specimens, and several are no longer responding to communications.  Dr. Savage has proposed a strategic redeployment to the underground complex known as the VAULT, and continuing production of PE062 there."
     ]
   },
+  {
+    "type": "snippet",
+    "category": "radio_archive",
+    "text": [ 
+      "kssht.  Dark Horse, this is Blue Jay, what's your status, over.  kssht.  Blue Jay, this is Black Horse, still holding, but not for long.  kssht.  Dark Horse, you've got to hold position for 3 hours.  We're almost black on ammo, but resupply is on the way, over.  kssht.  Not possible Blue Jay, too many dead. 30 minutes max, and if you don't order us to retreat we're gone, over.  kssht.",
```
Comment (Night-Pryanik) [inline data/json/snippets.json]:
```suggestion
      "kssht.  Dark Horse, this is Blue Jay, what's your status, over.  kssht.  Blue Jay, this is Black Horse, still holding, but not for long.  kssht.  Dark Horse, you've got to hold position for 3 hours.  We're almost black on ammo, but resupply is on the way, over.  kssht.  Not possible Blue Jay, too many dead.  30 minutes max, and if you don't order us to retreat we're gone, over.  kssht.",
```

File: data/json/snippets.json [hunk 2]
```diff
@@ -2292,6 +2292,34 @@
...
+      "To whomever is listening, this may be our last broadcast.  Wish you luck.  Can't stay in the studio any longer, station is being rewired to military frequencies for automatic broadcast.  Stay safe, and bless you, people.",
+      "You're listening to KDDA from Boston, I'm Jenny Sanders with a special emergency news update.  Road closures by military and police forces have been reported on interstates 90, 91, 93, and 95. Alternative routes have not been suggested. Due to dangers on non-interstate roads, it is strongly recommended that you avoid major thoroughfares and population centers when travelling by car.",
+      "Those evacuating cities are advised to head to the nearest designated evacuation shelter and await pickup by a FEMA transport.  Do not engage with rioters. Due to the danger of being mistaken for a rioter, do not approach military or police blockades, even for assistance.  I repeat, do not approach military or police blockades, even for assistance.",
```
Comment (Night-Pryanik) [inline data/json/snippets.json]:
```suggestion
      "Those evacuating cities are advised to head to the nearest designated evacuation shelter and await pickup by a FEMA transport.  Do not engage with rioters.  Due to the danger of being mistaken for a rioter, do not approach military or police blockades, even for assistance.  I repeat, do not approach military or police blockades, even for assistance.",
```

File: data/json/snippets.json [hunk 3]
```diff
@@ -2292,6 +2292,34 @@
...
+      "If you are not in an evacuation zone, officials highly recommend you pack gear for evacuation. Make sure to include clean clothes, a blanket, and enough food and water to last a few days.",
```
Comment (Night-Pryanik) [inline data/json/snippets.json]:
```suggestion
      "If you are not in an evacuation zone, officials highly recommend you pack gear for evacuation.  Make sure to include clean clothes, a blanket, and enough food and water to last a few days.",
```

File: data/json/snippets.json [hunk 4]
```diff
@@ -2292,6 +2292,34 @@
...
+      "In other news, the federal government has confirmed that Rhode Island's declaration of secession is official, although the US government does not acknowledge its validity.  Our sources haven't been able to determine the extent of hostilities in the area, but there are rumors of border violence initiated by the Rhode Island militia.",
+      "Hey ladies and gentlemen of the apocalypse, you're listening to Dark Days Radio, the last radio station on the Eastern Seaboard, I'm DJ Dustbowl here with our mascot Sam.",
+      "DJ Dustbowl back, you're listening to Dark Days Radio. Quick survivor tip: remember, killing a zombie isn't enough. You've gotta smash that thing to a pulp or chop it into pieces if you don't want it getting back up.  Good news though, you don't need to worry about headshots!  Any major damage will lay a shambler out.",
```
Comment (Night-Pryanik) [inline data/json/snippets.json]:
```suggestion
      "DJ Dustbowl back, you're listening to Dark Days Radio.  Quick survivor tip: remember, killing a zombie isn't enough. You've gotta smash that thing to a pulp or chop it into pieces if you don't want it getting back up.  Good news though, you don't need to worry about headshots!  Any major damage will lay a shambler out.",
```

File: data/json/snippets.json [hunk 5]
```diff
@@ -2292,6 +2292,34 @@
...
+      "You're listening to Dark Days Radio, I'm DJ Dustbowl! In with a quick tip for any survivors listening: one zombie is bad news, two are worse.  Don't go Rambo on me.  Tackle threats in small numbers or through bottlenecks.  You aren't tougher than a zombie but you are smarter, use that to your advantage... or just run away.",
```
Comment (Night-Pryanik) [inline data/json/snippets.json]:
```suggestion
      "You're listening to Dark Days Radio, I'm DJ Dustbowl!  In with a quick tip for any survivors listening: one zombie is bad news, two are worse.  Don't go Rambo on me.  Tackle threats in small numbers or through bottlenecks.  You aren't tougher than a zombie but you are smarter, use that to your advantage... or just run away.",
```

File: data/json/snippets.json [hunk 6]
```diff
@@ -2292,6 +2292,34 @@
...
+      "…black goo in the water!  Drugs and who knows what else, the govenment wants you dumb and docile!  Wake up people.  They are the spawn of Satan!  This is Alan Jewels with the AJ show, we’ll be back after the break.",
```
Comment (Night-Pryanik) [inline data/json/snippets.json]:
```suggestion
      "…black goo in the water!  Drugs and who knows what else, the government wants you dumb and docile!  Wake up people.  They are the spawn of Satan!  This is Alan Jewels with the AJ show, we’ll be back after the break.",
```

File: data/json/snippets.json [hunk 7]
```diff
@@ -2292,6 +2292,34 @@
...
+      "This is Escort, frequency one-fifty-five, nineteen-thousand, here's your report. Can you gimme an 'amen', it's ...kshhhsk.... days after Armageddon and I'm still kicking.  The 49ers are headed southwards towards Neufolk, so if you're there and listening, barricade or lay low until they go by.  There's another horde forming out of Metro Bay.",
```
Comment (Night-Pryanik) [inline data/json/snippets.json]:
```suggestion
      "This is Escort, frequency one-fifty-five, nineteen-thousand, here's your report.  Can you gimme an 'amen', it's ...kshhhsk.... days after Armageddon and I'm still kicking.  The 49ers are headed southwards towards Neufolk, so if you're there and listening, barricade or lay low until they go by.  There's another horde forming out of Metro Bay.",
```

File: data/json/snippets.json [hunk 8]
```diff
@@ -2292,6 +2292,34 @@
...
+      "This is not a test.  Seek immediate shelter.  Multiple missile launches have been confirmed to be targeting the ..ksshhsk... Seek immediate shelter.  If a community shelter is not available, alternative shelter locations are basements, beneath stairwells, or central rooms with no windows.  Ensure that you have sufficient protection from falling debris.  Ensure that you have food and water for at least one week.  Repeat.  Seek immediate shelter.",
+      "Hello American patriots!  This is the national guard, calling you to use your God-Given second amendment rights and buy a firearm!  It is obvious that the Chinese, a scourge upon the face of the earth and the stopper of freedom and prosperity, are in a state of aggression.  Therefore, we suggest that with the new gun laws introduced, YOU must help in the defense of this bastion of freedom and prosperity. Go to your local gun store and prepare to defend yourself against the red scourge in every way you can!",
```
Comment (Night-Pryanik) [inline data/json/snippets.json]:
```suggestion
      "Hello American patriots!  This is the national guard, calling you to use your God-Given second amendment rights and buy a firearm!  It is obvious that the Chinese, a scourge upon the face of the earth and the stopper of freedom and prosperity, are in a state of aggression.  Therefore, we suggest that with the new gun laws introduced, YOU must help in the defense of this bastion of freedom and prosperity.  Go to your local gun store and prepare to defend yourself against the red scourge in every way you can!",
```

File: data/json/snippets.json [hunk 9]
```diff
@@ -2292,6 +2292,34 @@
...
+      "zzzzchzzzzchurzzz Are you tired of always losing your dumpy old sunglasses?  Tired of having to pay for over price shades just to have them fall to pieces?  Well no longer now introducing zchzzzzch- brand Anti-glare compensators for the shade you need in you.  Order now for 12 easy payments of $1200.99 and we'll throw in a free order of zpzzzzpzz eye bot.",
+      "This is WSSA-233, broadcasting from (some city in the player's reality with a radio station).  We've barricaded all the doors and windows.  There's a ton of them right outside, we're trying to stay quiet.  Do not, I repeat, do not approach.",
```
Comment (Night-Pryanik) [inline data/json/snippets.json]:
```suggestion
      "This is WSSA-233, broadcasting from <city>.  We've barricaded all the doors and windows.  There's a ton of them right outside, we're trying to stay quiet.  Do not, I repeat, do not approach.",
```

File: data/json/snippets.json [hunk 10]
```diff
@@ -2292,6 +2292,34 @@
...
+      "First Fleet, this is Mount Wheater, we've send a helicopter with a squad to the location of The Guard Outpost. If any survivors of the crash are to reach the outpost, contact the base, we are short on men and one of the big ones has just breached the perimeter. Over.",
```
Comment (Night-Pryanik) [inline data/json/snippets.json]:
```suggestion
      "First Fleet, this is Mount Wheater, we've send a helicopter with a squad to the location of the Guard Outpost. If any survivors of the crash are to reach the outpost, contact the base, we are short on men and one of the big ones has just breached the perimeter.  Over.",
```

File: data/json/snippets.json [hunk 11]
```diff
@@ -2292,6 +2292,34 @@
...
+      "USS Eagle of Freedom to all callsigns.  Operation Ocean 11 is in effect.  Repeat.  Operation Ocean 11 is in effect.  Sir, this should effectivly reach about 150 units, but there is no response from literally anyone in hours.  Their coms winked out like candles in the wind.  And how do we even secure the port to load anyone in?",
```
Comment (Night-Pryanik) [inline data/json/snippets.json]:
```suggestion
      "USS Eagle of Freedom to all callsigns.  Operation Ocean 11 is in effect.  Repeat.  Operation Ocean 11 is in effect.  Sir, this should effectively reach about 150 units, but there is no response from literally anyone in hours.  Their coms winked out like candles in the wind.  And how do we even secure the port to load anyone in?",
```

File: data/json/snippets.json [hunk 12]
```diff
@@ -2292,6 +2292,34 @@
...
+      "Seagull 54 to USS Eagle of Freedom.  It's as we predicted.  Going down on last reserves of fuel, with dropping evacuated unit out of question at that point.  This is our last call.  Will try energency landing at ...kshssssssk... Don't send us a rescue team, we know the odds.  Will be moving east towards the shore, and if we're not there in T minus 3 days, count us dead. Proud to serve the country. Over and out."
```
Comment (Night-Pryanik) [inline data/json/snippets.json]:
```suggestion
      "Seagull 54 to USS Eagle of Freedom.  It's as we predicted.  Going down on last reserves of fuel, with dropping evacuated unit out of question at that point.  This is our last call.  Will try emergency landing at ...kshssssssk... Don't send us a rescue team, we know the odds.  Will be moving east towards the shore, and if we're not there in T minus 3 days, count us dead.  Proud to serve the country.  Over and out."
```

File: data/json/snippets.json (hunk 1 above)
Comment (tinukedaya) [reply data/json/snippets.json]: Blue Jay is calling **Dark** Horse, but it's **Black** Horse coming back.

File: data/json/snippets.json [hunk 13]
```diff
@@ -2296,28 +2296,28 @@
     "type": "snippet",
     "category": "radio_archive",
     "text": [ 
-      "kssht.  Dark Horse, this is Blue Jay, what's your status, over.  kssht.  Blue Jay, this is Black Horse, still holding, but not for long.  kssht.  Dark Horse, you've got to hold position for 3 hours.  We're almost black on ammo, but resupply is on the way, over.  kssht.  Not possible Blue Jay, too many dead. 30 minutes max, and if you don't order us to retreat we're gone, over.  kssht.",
+      "kssht.  Dark Horse, this is Blue Jay, what's your status, over.  kssht.  Blue Jay, this is Black Horse, still holding, but not for long.  kssht.  Dark Horse, you've got to hold position for 3 hours.  We're almost black on ammo, but resupply is on the way, over.  kssht.  Not possible Blue Jay, too many dead.  30 minutes max, and if you don't order us to retreat we're gone, over.  kssht.",
```
Comment (tinukedaya) [inline data/json/snippets.json]: Blue Jay is calling **Dark** Horse, but it's **Black** Horse coming back.

**Pull request merged**